news/management/commands/process_donations_queue.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import print_function, unicode_literals
+
+import json
+import sys
+from time import time
+
+from django.conf import settings
+from django.core.management import BaseCommand, CommandError
+
+import boto3
+import requests
+from django_statsd.clients import statsd
+
+from news.tasks import process_donation
+
+
+class Command(BaseCommand):
+    snitch_delay = 300  # 5 min
+    snitch_last_timestamp = 0
+    snitch_id = settings.DONATE_SNITCH_ID
+
+    def snitch(self):
+        if not self.snitch_id:
+            return
+
+        time_since = int(time() - self.snitch_last_timestamp)
+        if time_since > self.snitch_delay:
+            requests.post('https://nosnch.in/{}'.format(self.snitch_id))
+            self.snitch_last_timestamp = time()
+
+    def handle(self, *args, **options):
+        if not settings.DONATE_ACCESS_KEY_ID:
+            raise CommandError('AWS SQS Credentials not configured')
+
+        sqs = boto3.resource('sqs',
+                             region_name=settings.DONATE_QUEUE_REGION,
+                             aws_access_key_id=settings.DONATE_ACCESS_KEY_ID,
+                             aws_secret_access_key=settings.DONATE_SECRET_ACCESS_KEY)
+        queue = sqs.Queue(settings.DONATE_QUEUE_URL)
+
+        try:
+            # Poll for messages indefinitely.
+            while True:
+                self.snitch()
+                msgs = queue.receive_messages(WaitTimeSeconds=settings.DONATE_QUEUE_WAIT_TIME,
+                                              MaxNumberOfMessages=10)
+                for msg in msgs:
+                    if not (msg and msg.body):
+                        continue
+
+                    statsd.incr('mofo.donations.message.received')
+                    try:
+                        process_donation.delay(json.loads(msg.body))
+                    except Exception as e:
+                        statsd.incr('mofo.donations.message.error')
+                        print('ERROR:', e, '::', msg.body)
+                        continue
+                    else:
```

Comment:
If you use `else` in case of an exception you'd keep the msg in the queue which it can eventually result in receiving MaxNumberOfMessages all raising exceptions from the queue, thus blocking the processing. Maybe change this to `finally`?